pkg/services/datasources/service/secrets_mig.go

```diff
@@ -0,0 +1,92 @@
+package service
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/infra/kvstore"
+	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+)
+
+const (
+	secretMigrationStatusKey       = "secretMigrationStatus"
+	compatibleSecretMigrationValue = "compatible"
+	completeSecretMigrationValue   = "complete"
+)
+
+type DataSourceSecretMigrationService struct {
+	dataSourcesService datasources.DataSourceService
+	kvStore            *kvstore.NamespacedKVStore
+	features           featuremgmt.FeatureToggles
+}
+
+func ProvideDataSourceMigrationService(
+	dataSourcesService datasources.DataSourceService,
+	kvStore kvstore.KVStore,
+	features featuremgmt.FeatureToggles,
+) *DataSourceSecretMigrationService {
+	return &DataSourceSecretMigrationService{
+		dataSourcesService: dataSourcesService,
+		kvStore:            kvstore.WithNamespace(kvStore, 0, secretType),
```

Comment:
This comment is to clarify how kvStore is being used in this migration.

The kvStore is being used to store the secretMigrationStatus key, which is global, meaning we only have one for every org, that's why the orgId is 0.


That key can have 3 values:
Not set -> Migration never happened
“compatible” -> Migration happened with disableSecretCompatibility set to false
“complete” -> Migration happened with disableSecretCompatibility set to true